### PR TITLE
Handle class docstrings 

### LINKED ISSUE #1597: Missing Class Docstring
### Describe the bug
The docstring of a class is ignored. This means that it is not printed in the generated code, or in the wrapper.

### To Reproduce
Provide code to reproduce the behavior:
```python
class A:
    """
    Empty class
    """
    def __init__(self):
        pass
```

### Error details
Provide the generated code, or the error message:
```
ERROR at parsing (syntax) stage
pyccel:
 |error [syntax]: tmp10.py [2,4]| <class 'pyccel.ast.core.CommentBlock'> not currently supported in classes (<pyccel.ast.core.CommentBlock object at 0x7fd616054720>)

```

### Expected behavior
The docstring should be printed above the class definition.

### PR DESCRIPTION
**Pull request by EmilyBourne**
Add the necessary code to print class docstrings and make them available in Python. Fixes #1597

### CODE REVIEW

File: pyccel/ast/bind_c.py [hunk 1]
```diff
@@ -553,12 +553,15 @@ class BindCClassDef(ClassDef):
...
+    **kwargs : dict
+        See ClassDef.
     """
     __slots__ = ('_original_class',)
 
-    def __init__(self, original_class):
+    def __init__(self, original_class, **kwargs):
         self._original_class = original_class
-        super().__init__(original_class.name)
+        super().__init__(original_class.name, **kwargs)
```
Comment (yguclu) [inline pyccel/ast/bind_c.py:564]: Pyccel Bot complains that `BindCClassDef` is not part of `__all__`

File: pyccel/ast/core.py [hunk 2]
```diff
@@ -3184,6 +3184,9 @@ class ClassDef(ScopedAstNode):
     interfaces : iterable
         The interface methods.
 
+    doc_string : CommentBlock, optional
+        The doc string of the class.
```
Comment (yguclu) [inline pyccel/ast/core.py:3188]: Why don't you just call it `docstring` without the underscore?

File: pyccel/ast/cwrapper.py [hunk 3]
```diff
@@ -518,15 +518,18 @@ class definition.
...
     __slots__ = ('_original_class', '_struct_name', '_type_name', '_type_object')
 
-    def __init__(self, original_class, struct_name, type_name):
+    def __init__(self, original_class, struct_name, type_name, **kwargs):
         self._original_class = original_class
         self._struct_name = struct_name
         self._type_name = type_name
         self._type_object = Variable(PyccelPyClassType(), type_name)
-        super().__init__(original_class.name)
+        super().__init__(original_class.name, **kwargs)
```
Comment (yguclu) [inline pyccel/ast/cwrapper.py:532]: Pyccel Bot complains that `PyClassDef` is missing from `__all__`

File: pyccel/ast/core.py (hunk 2 above)
Comment (EmilyBourne) [reply pyccel/ast/core.py:3188]: I'm glad you said this. I kept making typos and writing `docstring`. I chose `doc_string` to match what is in `FunctionDef`. Can I go ahead and change both?

**Review by yguclu — APPROVED** [review]

File: pyccel/ast/bind_c.py (hunk 1 above)
Comment (EmilyBourne) [reply pyccel/ast/bind_c.py:565]: Fixed [d7d71f7](https://github.com/pyccel/pyccel/pull/1620/commits/d7d71f7d00edd6eed967ad3c5874f1aca51f4018)

File: pyccel/ast/core.py (hunk 2 above)
Comment (yguclu) [reply pyccel/ast/core.py:3224]: Sure

File: pyccel/ast/cwrapper.py (hunk 3 above)
Comment (yguclu) [reply pyccel/ast/cwrapper.py:535]: Fixed in d7d71f7

File: pyccel/ast/bind_c.py (hunk 1 above)
Comment (yguclu) [reply pyccel/ast/bind_c.py:565]: Fixed in d7d71f7

File: pyccel/ast/core.py (hunk 2 above)
Comment (EmilyBourne) [reply pyccel/ast/core.py]: Done [78adce6](https://github.com/pyccel/pyccel/pull/1620/commits/78adce6893e5aa11292acee961434e370bf26e29)

**Pull request merged**